spec/features/user_logs_in_with_facebook_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+require 'rails_helper'
+
+describe "user logs in facebook" do
+
+  before do
+    OmniAuth.config.test_mode = true
+    OmniAuth.config.add_mock(:facebook, {
+    uid: "1234",
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.